json_parser.cpp

```diff
@@ -67,148 +67,156 @@ class Valuesax {
 };
 
 Valuesax Valuesax::Parse(std::string const& s) {
-    Valuesax result;
-    result._type = Valuesax::String;
+  Valuesax result;
+  result._type = Valuesax::String;
 
-    result._string = s;
-    return result;
+  result._string = s;
+  return result;
 }
 
 Valuesax Valuesax::Parse(const double& s) {
-    Valuesax result;
-    result._type = Valuesax::Double;
+  Valuesax result;
+  result._type = Valuesax::Double;
 
-    result._double = s;
-    return result;
+  result._double = s;
+  return result;
 }
 
 Valuesax Valuesax::Parse(bool& s) {
-    Valuesax result;
-    result._type = Valuesax::Bool;
+  Valuesax result;
+  result._type = Valuesax::Bool;
 
-    result._bool = s;
-    return result;
+  result._bool = s;
+  return result;
 }
 
 Valuesax Valuesax::Parse(const unsigned& s) {
-    Valuesax result;
-    result._type = Valuesax::Decimal;
+  Valuesax result;
+  result._type = Valuesax::Decimal;
 
-    result._num = s;
-    return result;
+  result._num = s;
+  return result;
 }
 
 Valuesax Valuesax::Parse(const std::nullptr_t& s) {
-    Valuesax result;
-    result._type = Valuesax::Null;
+  Valuesax result;
+  result._type = Valuesax::Null;
 
-    result._null = s;
-    return result;
+  result._null = s;
+  return result;
 }
 
 class MyHandler : public BaseReaderHandler<UTF8<>, MyHandler> {
-    public:
+  public:
     std::vector < std::pair < std::string, Valuesax>>  mymap;
     std::vector <char> vect;
     std::string keyname{"first"};
     std::string keyvalue;
     bool valuep{false};
     int start_counter{0};
     std::vector<std::string> mystack;
-    std::unordered_map<int, std::string> key_stack;
+    std::unordered_map<int, std::string> key_stack;//TODO why not using an array?
```

Comment:
why not using an array?

Reply:
the access is random and limited by the depth size of the JSON, quite ideal for stl::vector.
the vector(array) could initialize with some "big" size, let's say 400.
the vector can be resized upon index is too big (it's not happening too much)

Reply:
Yes, the access is random. But, assignment in this case, I think the start_counter values are not contiguous